src/hooks/useLocalStorage.ts

```diff
@@ -0,0 +1,16 @@
+import { useState } from 'react';
+
+type Value = string | number | object;
+
+export const useLocalStorage = (key: string, initialValue: Value) => {
+  const [value, setValue] = useState(
+    JSON.parse(localStorage.getItem(key) || JSON.stringify(initialValue)),
+  );
+
+  const save = (newValue: Value) => {
```

Comment:
For ECMAScript 5 compatibility, consider using 'function' instead of the arrow function. 🛠️